Docs: Add CSF factory API documentation (#30343)
storybookjs/storybook: docs/_snippets/portable-stories-csf-factory-render.md

```tsx filename="Button.test.tsx" renderer="react" language="ts"
import { test, expect } from 'vitest';
import { render, screen } from '@testing-library/react';

// Import all stories from the stories file
import * as stories from './Button.stories';

const { Primary, Secondary } = stories;

test('renders primary button with default args', async () => {
  // Access the story's component via the .Component property
  render(<Primary.Component />);
  const buttonElement = screen.getByText('Text coming from args in stories file!');
  expect(buttonElement).not.toBeNull();
});

test('renders primary button with overridden props', async () => {

>   // You can override props by passing them in the context argument of the play function

Author: ```suggestion
  // You can override props by passing them directly to the story's component
```
